### PR TITLE
Adds nightly builds.

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
Adds experimental nightly builder. More in ticket: #818

Squash merge recommended.

### CODE REVIEW

**Comment by davidhutchens** [comment]
I offer the following slight wording changes for the README Nightly Builds section:

We also offer nightly builds based on the current state of the
[develop](https://github.com/logisim-evolution/logisim-evolution/tree/develop) branch.
Note that these builds contain recent changes that may not be fully tested. 
Nightly builds are updated at midnight [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time).

**[Click here for a list of downloadable artifacts](https://github.com/logisim-evolution/logisim-evolution/actions/workflows/nightly.yml).**  

Please share your experiences in [Discussions](https://github.com/logisim-evolution/logisim-evolution/discussions)
or [open a ticket](https://github.com/logisim-evolution/logisim-evolution/issues)
if you found a bug or have ideas for improvement.

**Comment by davidhutchens** [comment]
Overall, this looks good. But is there a reason that the nightly build needs to compile all source and build the shadowJar twice?

**Comment by MarcinOrlowski** [comment]
For some reason you seem to have proofread older version of the text. Can you please check the [recent one](https://github.com/MarcinOrlowski/logisim-evolution/blob/nightly-builds/README.md)?

As for builds - it's even worse as of now (but that's still WiP so that's my excuse :) I had to run `./gradlew clean` after each build as I had some problems with task `createDistDir` failing, complaining about more than one shadowJar files. I haven't had time to investigate yet.

**Comment by davidhutchens** [comment]
The check for multiple shadowJar files is supposed to trap if old versions were left hanging around. The Java jpackage program puts them all in the package. A single clean and consistent versions for building should not generate more than one. I'm not sure what caused more than one for you. But the test is needed since including multiple versions causes some problems.

**Comment by davidhutchens** [comment]
Hmmm. Perhaps you have the source jar in there?

**Comment by davidhutchens** [comment]
My mistake on the proofread. The first and third paragraphs above still work. Change your middle paragraph by inserting a "the" between "to" and "last". Does that make sense? Is there a better way for me to provide edits?

**Comment by MarcinOrlowski** [comment]
> Perhaps you have the source jar in there

Makes sense. I rearranged building steps moving source JAR to the end, so now it all builds with just one clean in the middle.

>  Is there a better way for me to provide edits?

For shorter texts it's probably less problematic to just post edited version. If you need to point to specific places then you can go to 
"Files changed" tab on the pull request, find the right place, hoover your mouse over the line you want and then click "+" to comment that exact line:

File: README.md [hunk 1]
```diff
@@ -74,6 +77,19 @@ include the Java runtime and do not require it to be installed separately:
...
 is also available and can be run on any system with a supported Java runtime installed.
 
+### Nightly builds ###
+
+We also offer builds based on current state of [develop](https://github.com/logisim-evolution/logisim-evolution/tree/develop)
+branch, which are created everyday at midnight [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time).
+
+To get nightly downloads, please [click here](https://github.com/logisim-evolution/logisim-evolution/actions/workflows/nightly.yml)
+and browse to the last successful run of "Nightly build" (should be on top). Note, that due to Github internals,
+all files are provided as ZIP archives. You must unpack unpack downloaded binary prior run or installation.
```
Comment (maehne) [inline README.md:87]: "unpack unpack" -> "unpack"

**Pull request merged**